examples/anvil/examples/deploy_contract_anvil.rs

```diff
@@ -0,0 +1,98 @@
+//! Example of deploying a contract to Anvil and interacting with it.
+
+use alloy_network::{Ethereum, EthereumSigner};
+use alloy_node_bindings::Anvil;
+use alloy_primitives::{U256, U64};
+use alloy_provider::{Provider, ProviderBuilder, RootProvider};
+use alloy_rpc_client::RpcClient;
+use alloy_rpc_trace_types::parity::TraceOutput;
+use alloy_signer::LocalWallet;
+use alloy_sol_types::sol;
+use alloy_transport_http::Http;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Spin up a local Anvil node.
+    // Ensure `anvil` is available in $PATH
+    let anvil = Anvil::new().spawn();
+
+    // Set up wallet
+    let wallet: LocalWallet = anvil.keys()[0].clone().into();
+
+    // Create a provider with a signer and the network.
+    let http = Http::new(anvil.endpoint().parse()?);
+    let provider = ProviderBuilder::<_, Ethereum>::new()
+        .signer(EthereumSigner::from(wallet))
+        .network::<Ethereum>()
+        .provider(RootProvider::new(RpcClient::new(http, true)));
+
+    println!("Anvil running at `{}`", anvil.endpoint());
+
+    // Create a contract.
+    sol! {
+        // solc v0.8.24; solc a.sol --via-ir --optimize --bin
+        #[sol(rpc, bytecode="608080604052346100155760d2908161001a8239f35b5f80fdfe60808060405260043610156011575f80fd5b5f3560e01c9081633fb5c1cb1460865781638381f58a14606f575063d09de08a146039575f80fd5b34606b575f366003190112606b575f545f1981146057576001015f55005b634e487b7160e01b5f52601160045260245ffd5b5f80fd5b34606b575f366003190112606b576020905f548152f35b34606b576020366003190112606b576004355f5500fea2646970667358221220bdecd3c1dd631eb40587cafcd6e8297479db76db6a328e18ad1ea5b340852e3864736f6c63430008180033")]
+        contract Counter {
+            uint256 public number;
+
+            function setNumber(uint256 newNumber) public {
+                number = newNumber;
+            }
+
+            function increment() public {
+                number++;
+            }
+        }
+    }
+
+    // Get the base fee for the block.
+    let base_fee = provider.get_gas_price().await?;
+
+    // Deploy the contract.
+    let contract_builder = Counter::deploy_builder(&provider);
+    let estimate = contract_builder.estimate_gas().await?;
+    let pending = contract_builder
+        .gas(estimate)
+        .gas_price(base_fee)
+        .nonce(U64::from(0))
```

Comment:
Would be great if `gas`, `gas_price` are derived automatically.

Once I've figured out the nonce manager I'll update the examples so they use it.

Reply:
Does it not work without? Gas doesn't have to be present in the request as the node will fill it

Reply:
If I don't include the `gas` I get: `TransportError(Transport(Custom(MissingKey("gas_limit"))))`

If I don't include the `gas_price` I get: `TransportError(Transport(Custom(MissingKey("max_priority_fee_per_gas"))))`

Reply:
this is because of the signer layer @DaniPopes, the node doesn't fill it because it is getting a signed tx (we sign it locally)

if you remove the signer layer, it works